### DISCUSSION TITLE
protoc import "cue/cue.proto"; ?

### DISCUSSION THREAD

**Discussion by cueckoo** [cue-lang/cue#983] [General] [answered]
_Originally opened by @randomizedcoder in https://github.com/cuelang/cue/discussions/983_

G'day,

I'm a newbie attempting to use Cue for the first time, so sorry for the really basic questions. I figure I should ask to help the next newbie, and possibly add a little to the docs to make it more clear. - Documentation does seems great BTW

Essentially, I'm hoping to this CUE mostly to code gen input validation for gRPC, per the tail end of:
https://cuelang.org/docs/usecases/generate/

I've added the cue tags to the .protos, per:
https://cuelang.org/docs/integrations/protobuf/#extract-cue-from-a-standalone-proto-file

I've done the install via: https://cuelang.org/docs/install/
Added the import for cue, just after the standard timestamp, for example:

```
import "google/protobuf/timestamp.proto";
import "cue/cue.proto";
```

Then I tried protoc, but I suspect I'm missing a plugin.

```
/bin/protoc *.proto --go_out=plugins=grpc:/my/go/src/
/bin/protoc *.proto --go_out=plugins=grpc:/home/das/go/src/
cue/cue.proto: File not found.
mp.proto: Import "cue/cue.proto" was not found or had errors.
```

But I don't know what to do.  Searching finds the envoyproxy stuff, and actually that might be an alternative to cue.  I don't know what the community is doing, with CUE verse this envoy proxy code gen thinger.

https://github.com/envoyproxy/protoc-gen-validate

Thanks in advance

**Comment by myitcv** [comment]
@randomizedcoder I'm very sorry, your original message entirely slipped through the gaps.

Are you still having problems here? If so, please respond back and I'll be happy to help.

**Comment by randomizedcoder** [comment, marked as answer]
I could not get cue working because of the issue I posed.

On Mon, Jul 26, 2021 at 2:25 AM Paul Jolly ***@***.***> wrote:

> @randomizedcoder <https://github.com/randomizedcoder> I'm very sorry,
> your original message entirely slipped through the gaps.
>
> Are you still having problems here? If so, please respond back and I'll be
> happy to help.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cue-lang/cue/discussions/983#discussioncomment-1050306>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APMCHTSNU5KNHD7G2G544XLTZUS2PANCNFSM47YASLRQ>
> .
>


-- 
Regards,
Dave Seddon
+1 415 857 5102

**Reply by myitcv** [reply]
@randomizedcoder the issue here is that `protoc` does not know where to find `cue/cue.proto`. Hence you need to pass `-I` to `protoc` to indicate where imports can be resolved. As far as I'm aware, the best way to do that is to establish a dependency on the `cuelang.org/go` Go module which contains that file. For a complete example please see https://github.com/cue-lang/cue/issues/581#issuecomment-891926335, a comment which unfortunately also confirms there is a problem with the Go code that is generated today 😢 ! But #581 tracks fixing that.

**Comment by randomizedcoder** [comment]
Thanks Paul.  I feel less silly now. :)  I guess this item can be closed as
a duplicate.

On Tue, Aug 3, 2021 at 8:16 AM Paul Jolly ***@***.***> wrote:

> @randomizedcoder <https://github.com/randomizedcoder> the issue here is
> that protoc does not know where to find cue/cue.proto. Hence you need to
> pass -I to protoc to indicate where imports can be resolved. As far as
> I'm aware, the best way to do that is to establish a dependency on the
> cuelang.org/go Go module which contains that file. For a complete example
> please see #581 (comment)
> <https://github.com/cue-lang/cue/issues/581#issuecomment-891926335>, a
> comment which unfortunately also confirms there is a problem with the Go
> code that is generated today 😢 ! But #581
> <https://github.com/cue-lang/cue/issues/581> tracks fixing that.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/cue-lang/cue/discussions/983#discussioncomment-1125698>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APMCHTSZEIWQCFUARRBBOYLT3AB5ZANCNFSM47YASLRQ>
> .
> Triage notifications on the go with GitHub Mobile for iOS
> <https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675>
> or Android
> <https://play.google.com/store/apps/details?id=com.github.android&utm_campaign=notification-email>
> .
>


-- 
Regards,
Dave Seddon
+1 415 857 5102

**Reply by myitcv** [reply]
@randomizedcoder honestly don't feel silly even in the slightest! This is certainly not clear or obvious from the documentation, and on top of that there is a bug/missing feature. 

Any issue/question is honestly an opportunity for us to learn how things can be improved. So thank you for asking the question in the first place.
